pkg/plugin/provision/spirehelm/providerfactory.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Cofide Limited.
+// SPDX-License-Identifier: Apache-2.0
+
+package spirehelm
+
+import (
+	"context"
+
+	trust_zone_proto "github.com/cofide/cofide-api-sdk/gen/go/proto/trust_zone/v1alpha1"
+
+	"github.com/cofide/cofidectl/pkg/plugin"
+	"github.com/cofide/cofidectl/pkg/provider/helm"
+)
+
+// Type check that HelmSPIREProviderFactory implements the ProviderFactory interface.
+var _ ProviderFactory = &HelmSPIREProviderFactory{}
+
+// ProviderFactory is an interface that abstracts the construction of HelmSPIREProvider objects.
+type ProviderFactory interface {
+	// Build returns a HelmSPIREProvider configured with values for an install/upgrade.
```

Comment:
Very minor, but if we forsee `ProviderFactory` being an interface equivalent to `Provider`, we'll want to remove the `HelmSPIRE*` part here from this comment (provided I'm understanding correctly)

Reply:
Good point, will update.